package.json

```diff
@@ -35,7 +35,7 @@
     "prettier:check": "prettier --check \"**/*.{js,ts,tsx,json,md}\"",
     "test": "jest --env jsdom",
     "ts": "tsc --build tsconfig.json",
-    "prepublishOnly": "yarn lint && yarn test && yarn build",
+    "prepublishOnly": "yarn lint && yarn test && yarn build && yarn ts",
```

Comment:
Previously, only new JS files would be built as a part of the pre-publish script, leaving potentially (though unlikely) out of date TS files in the `dist` directory of packages being published.

Reply:
Should we swap the order of this then? Cuz yarn lint already runs yarn ts.

Reply:
Whoops, seemed to have forgotten about that. Should we actually have lint run tsc without emitting declaration files? It seems like a very side effect-y thing to be doing for a linting script.

Reply:
If we can do that easily then that WFM but it may be tricky to actually pull off.

Might be easier to just remove it from lint entirely and have it be its own individual task that we call individually?

Reply:
I tried just passing `--noEmit`, but that didn't work out the box. I wanna keep this on our radar as an improvement we could make, because I feel pretty strongly relying on a lint script to build part of our packages is a bad idea. I'll make a ticket.

Reply:
https://jira.mongodb.org/browse/PD-603